src/main/app/document/DocumentManagementController.ts

```diff
@@ -420,21 +422,45 @@ export class DocumentManagerController extends PostController<AnyObject> {
     let uid = '';
     let documentToGet = '';
     let ele = '';
-    const document_filename = req.session.userCase[`${element}`]?.document_filename;
+    let ele1 = '';
+    let document_filename = req.session.userCase[`${element}`]?.document_filename;
 
     if (element !== null && element !== undefined) {
       ele = element[0];
-      if (!req.session.userCase[`${ele}`]?.document_binary_url) {
-        throw new Error('binary url is not found for ' + document_filename);
-      }
-      documentToGet = req.session.userCase[`${ele}`]?.document_binary_url;
-      uid = this.getUID(documentToGet);
+      ele1 = element[1];
+      if (ele !== 'respondentDocsList') {
+        if (!req.session.userCase[`${ele}`]?.document_binary_url) {
+          throw new Error('binary url is not found for ' + document_filename);
+        }
+        documentToGet = req.session.userCase[`${ele}`]?.document_binary_url;
+        uid = this.getUID(documentToGet);
 
-      if (flag !== null || flag !== undefined) {
-        flag = YesOrNo.YES;
+        if (flag !== null || flag !== undefined) {
+          flag = YesOrNo.YES;
+        }
+      } else {
+        if (ele1 === 'c1a') {
+          for (const document of req.session.userCase.respondentDocsList!) {
```

Comment:
you can refactor the two for loops to one by handing conditions inside of the loop